src/utils/PipeHelpers.ts

```diff
@@ -0,0 +1,40 @@
+// tslint:disable no-console
+
+import ParallelTransform = require('parallel-transform');
+import stream = require('stream');
+import { unpromisify } from './utils';
+
+export const parallelTransform = (concurrency: number, handler: (data: any) => Promise<any>): stream.Transform =>
+  new ParallelTransform(concurrency, {
+  objectMode: true,
+}, async (data, callback) => {
+  unpromisify(async () => await handler(data), callback);
+});
+
+export const readableStream = (handler: () => Promise<any>): stream.Readable =>
+  new stream.Readable({
+    objectMode: true,
+    async read() {
+      let value;
+      let error;
+      try {
+        value = await handler();
+      } catch (e) {
+        error = e;
+        console.error(`Error reading from stream: ${error}`);
+      }
+      if (value) {
+        this.push(value);
```

Comment:
what is this pushing into?

Reply:
This is just the streams API for read streams.  Read is called on you, and you respond with this.push.
https://nodejs.org/api/stream.html#stream_readable_streams